.schemas/queries.json

```diff
@@ -0,0 +1,262 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "type": "object",
+    "properties": {
+        "metadata": {
+            "$ref": "#/$defs/metadata"
+        },
+        "defaults": {
+            "type": "object",
+            "properties": {
+                "metadata": {
+                    "type": "object",
+                    "properties": {
+                        "data_source": {
+                            "type": "string"
+                        },
+                        "data_families": {
+                            "type": "array",
+                            "items": {
+                                "type": "string"
+                            }
+                        },
+                        "pivot": {
+                            "type": "object",
+                            "patternProperties": {
+                                ".*": {
+                                    "type": "array",
+                                    "items": {
+                                        "type": "string"
+                                    }
+                                }
+                            }
+                        },
+                        "top": {
+                            "type": "integer"
+                        }
+                    }
+                },
+                "parameters": {
+                    "$ref": "#/$defs/parameter"
+                }
+            }
+        },
+        "sources": {
+            "type": "object",
+            "patternProperties": {
+                ".*": {
+                    "$ref": "#/$defs/query"
+                }
+            }
+        }
+    },
+    "$defs": {
+        "description": {
+            "type": "string",
+            "minLength": 5,
+            "maxLength": 256
+        },
+        "metadata": {
+            "type": "object",
+            "properties": {
+                "version": {
+                    "type": "integer"
+                },
+                "description": {
+                    "$ref": "#/$defs/description"
+                },
+                "data_environments": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "enum": [
```

Comment:
Is it possible to have this list open-ended? E.g. for custom providers.
Maybe not that important for our use of the json file.
It's also missing a couple of recent providers - MSSentinel_New and Kusto_New

Reply:
The list can be open ended thanks to the "anyOf" as explained here > https://github.com/json-schema-org/json-schema-spec/issues/665
However that effectively means the list of valid providers will only be there for autocompletion when integrated with vscode (or another text editor).

I'll implement the change for this